cli/src/pcluster/resources/custom_resources/OpenSearch_lambda/Lambda_pushing_events_to_OpenSearch.py

```diff
@@ -103,11 +180,22 @@ def is_valid_json(message):
 
 def post(body):
     headers = {"Content-Type": "application/json"}
-    request = AWSRequest(method="POST", url=endpoint, data=body, headers=headers)
-    request.context[
-        "payload_signing_enabled"
-    ] = True  # This is mandatory since VpcLattice does not support payload signing.
+    endpoint_for_post = endpoint + "/_bulk"
+    request = AWSRequest(method="POST", url=endpoint_for_post, data=body, headers=headers)
+    request.context["payload_signing_enabled"] = True
+    sigv4.add_auth(request)
+    prepped = request.prepare()
+    response = requests.post(prepped.url, data=body, headers=prepped.headers, timeout=200)
+    return response
+
+
+def get(query):
+    data = json.dumps(query)
+    headers = {"Content-Type": "application/json"}
+    endpoint_for_get = endpoint + "/cwl-2023.07.*/_search"
+    request = AWSRequest(method="GET", url=endpoint_for_get, data=data, headers=headers)
+    request.context["payload_signing_enabled"] = True
     sigv4.add_auth(request)
     prepped = request.prepare()
-    response = requests.post(prepped.url, data=body, headers=prepped.headers, timeout=20)
+    response = requests.get(prepped.url, data=data, headers=prepped.headers, timeout=200)
```

Comment:
If you have time it might be nice to encapsulate the common functionality into a common function here, but not critical.

Reply:
Done